File: packages/dd-trace/test/llmobs/parity/scenarios/langchain/tool-invoke.py
from common import finish

from parity_langchain import chat_anthropic, chat_openai, openai_embeddings, start  # noqa: F401


start()
from langchain_core.tools import tool


@tool
def add(a: int, b: int) -> str:
    """Adds two numbers together"""
    return str(a + b)


add.invoke({"a": 2, "b": 2})
finish()

File: packages/dd-trace/test/llmobs/parity/scenarios/langchain/tool-invoke.js
'use strict'

const { finish } = require('../common')
const helpers = require('./helpers')

async function main () {
  const tracer = helpers.loadLangChainTracer()
  const { tool } = helpers.coreModule('@langchain/core/tools')
  const { z } = helpers.zod()
  const add = tool(({ a, b }) => String(a + b), {
    name: 'add',
    description: 'Adds two numbers together',
    schema: z.object({ a: z.number(), b: z.number() }),
  })
  await add.invoke({ a: 2, b: 2 })
  await finish(tracer)
}

main().catch(error => {
  process.stderr.write(`${error.stack}\n`)
  process.exitCode = 1
})
